File: 4. Trees and Graphs/breadthFirstSearch.js
const Node = function(value) {
    this.value = value;
    this.left = null;
    this.right = null;
}

const breadthFirstSearch = function(root, targetVal) {
    let queue = [root];
    let result = null;

    while(queue.length > 0) {
        let node = queue.shift();
        if(node.value === targetVal) {
            result = node;
        }
        if(node.left) {
            queue.push(node.left);
        }
        if(node.right){
            queue.push(node.right);
        }
    }
    return result;
}

File: 11. Common Algos/breadthFirstSearch.py
class Node:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

def breadthFirstSearch(root, targetVal):
    queue = [root]
    
    while len(queue) > 0:
        node = queue.pop(0)

        if node.val == targetVal:
            return node
        if node.left:
            queue.append(node.left)
        if node.right:
            queue.append(node.right)